caboose.go

```diff
@@ -136,19 +151,21 @@ func (c *Caboose) Close() {
 	c.logger.Close()
 }
 
-// Note: Caboose is NOT a persistent blockstore and does NOT have an in-memory cache. Every block read request will escape to the Saturn network.
-// Caching is left to the caller.
+// Fetch allows fetching car archives by a path of the form `/ipfs/<cid>[/path/to/file]`
+func (c *Caboose) Fetch(ctx context.Context, path string, cb DataCallback) error {
+	return c.pool.fetchResourceWith(ctx, path, cb, c.getAffinity(ctx))
```

Comment:
So this API will ONLY be used for fetching CARs, correct ? Can I give a path such as just /ipfs/{cid} to it ? 

Will that give me the entire CAR rooted at that cid or just the block  for that cid ? The Saturn URL in that case will be {Saturn Node IP}/ipfs/cid.  Does Saturn have an API in place already to be able to tell that it should return the entire CAR for that root and not just the block ? Is it the `format=raw?` part in `saturnReqTmpl` that indicates that we want to fetch just a block ?

Reply:
> So this API will ONLY be used for fetching CARs, correct ?
yes
> Can I give a path such as just /ipfs/{cid} to it ?
yes
> Will that give me the entire CAR rooted at that cid or just the block for that cid ? 
that's up to the backing saturn node to implement the semantics defined at
https://www.notion.so/pl-strflt/Graph-Request-APIs-97cc90e74df44404a0f9aa29d7291a5f